streamer/pulp/streamer/server.py

```diff
@@ -378,3 +376,60 @@ def write(self, data):
         :type  data: str
         """
         reactor.callFromThread(self.request.write, data)
+
+
+class SessionCache(Cache):
+    """
+    Session cache.
+    Extends generic cache to derive the key using the URL and
+    downloader configuration.
+    """
+
+    @staticmethod
+    def key(url, downloader):
+        """
+        Get a cache key for the downloader.
+
+        :param url: The download URL.
+        :type url: basestring
+        :param downloader: A file downloader.
+        :type downloader: nectar.downloaders.base.Downloader
+        :return: A cache key based on downloader configuration.
+        """
+        url = urlparse(url)
```

Comment:
Rather than re-useing the name and assigning it an object of a different type, consider using a new name such as ``url_parts``, ``parsed_url`` or similar.